src/Extensions/Diagnostics.HealthChecks/Internal/ServiceFabricHealthCheckPublisher.cs

```diff
@@ -34,61 +34,99 @@ internal class ServiceFabricHealthCheckPublisher : IHealthCheckPublisher
+			};
 
-				try
-				{
-					partition.ReportPartitionHealth(sfHealthInformation);
-				}
-				catch (Exception exception)
+			try
+			{
+				foreach (KeyValuePair<string, HealthReportEntry> entryPair in report.Entries)
 				{
-					m_logger.LogError(Tag.Create(), exception, "Failed to report partition health");
+					cancellationToken.ThrowIfCancellationRequested();
+					reportHealth(BuildSfHealthInformation(entryPair.Key, entryPair.Value));
 				}
+
+				cancellationToken.ThrowIfCancellationRequested();
+				reportHealth(BuildSfHealthInformation(report));
+			}
+			catch (FabricObjectClosedException)
+			{
+				// Ignore, the service instance is closing.
 			}
 
 			return Task.CompletedTask;
 		}
 
-		private string CreateDescription(HealthReportEntry entry)
+		private SfHealthInformation BuildSfHealthInformation(string healthCheckName, HealthReportEntry reportEntry)
 		{
-			// Healthy reports won't be displayed and most of the checks will be healthy,
-			// so we are creating a detailed description only for failed checks and avoiding allocations for healthy
-			if (entry.Status == HealthStatus.Healthy)
+			StringBuilder descriptionBuilder = new StringBuilder();
+			if (!string.IsNullOrWhiteSpace(reportEntry.Description))
 			{
-				return entry.Description;
+				descriptionBuilder.AppendLine($"Description: {reportEntry.Description}");
+			}
+			descriptionBuilder.AppendLine($"Duration: {reportEntry.Duration}.");
+			if (reportEntry.Exception != null)
+			{
+				descriptionBuilder.AppendLine($"Exception: {reportEntry.Exception}");
 			}
 
-			return new StringBuilder()
-				.Append("Exception:").AppendLine(entry.Exception?.ToString())
-				.Append("Description:").AppendLine(entry.Description)
-				.Append("Duration:").AppendLine(entry.Duration.ToString())
-				.Append("Tags:").AppendLine(string.Join(",", entry.Tags))
-				.ToString();
+			return new SfHealthInformation(SfHealthInformationSourceId, healthCheckName, ToSfHealthState(reportEntry.Status))
+			{
+				Description = descriptionBuilder.ToString(),
+			};
+		}
+
+		private SfHealthInformation BuildSfHealthInformation(HealthReport report)
+		{
+			int entriesCount = report.Entries.Count;
+			int healthyEntries = 0;
+			int degradedEntries = 0;
+			int unhealthyEntries = 0;
+			foreach (KeyValuePair<string, HealthReportEntry> entryPair in report.Entries)
+			{
+				switch (entryPair.Value.Status)
+				{
+					case HealthStatus.Healthy:
+						healthyEntries++;
+						break;
+					case HealthStatus.Degraded:
+						degradedEntries++;
+						break;
+					case HealthStatus.Unhealthy:
+					default:
+						unhealthyEntries++;
+						break;
+				}
+			}
+
+			StringBuilder descriptionBuilder = new StringBuilder()
+				.AppendLine($"Health checks executed: {entriesCount}. Healthy: {healthyEntries}/{entriesCount}. Degraded: {degradedEntries}/{entriesCount}. Unhealthy: {unhealthyEntries}/{entriesCount}.")
```

Comment:
I think it would be better to use `AppendFormated` (or something simular) instead, here and bellow, since using `$` would create additional `StringBuilder`for each case.

Reply:
Are you sure? https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated#compilation-of-interpolated-strings

Reply:
Yes, `StringFormat` build on top of `StringBuilder` inside, I had an application where replacing them significantly reduced allocations. Also, I've stumbled into this article https://www.meziantou.net/stringbuilder-performance-pitfalls.htm

Reply:
Changed to use .Append...